docs/updating/from_4.0/to_4.1.md

```diff
@@ -0,0 +1,251 @@
+        content_field_id integer not null,
+        version_no integer not null,
+        unit_id integer not null
+        constraint ibexa_measurement_value_unit_fk
+            references ibexa_measurement_unit
+            on update cascade on delete restrict,
+        value double precision not null,
+        constraint ibexa_measurement_value_attr_fk
+            foreign key (content_field_id, version_no) references ezcontentobject_attribute
+                on update cascade on delete cascade
+    );
+
+    CREATE INDEX IF NOT EXISTS ibexa_measurement_value_unit_id ON ibexa_measurement_value (unit_id);
+
+    CREATE UNIQUE INDEX IF NOT EXISTS ibexa_measurement_value_attr_ver ON ibexa_measurement_value (content_field_id, version_no);
+
+    CREATE UNIQUE INDEX IF NOT EXISTS ibexa_measurement_value_attr_ver_unit ON ibexa_measurement_value (content_field_id, version_no, unit_id);
+
+    CREATE TABLE IF NOT EXISTS ibexa_measurement_range_value
+    (
+        id serial primary key,
+        content_field_id integer not null,
+        version_no       integer not null,
+        unit_id          integer not null
+        constraint ibexa_measurement_range_value_unit_fk
+            references ibexa_measurement_unit
+            on update cascade on delete restrict,
+        min_value        double precision not null,
+        max_value        double precision not null,
+        constraint ibexa_measurement_range_value_attr_fk
+            foreign key (content_field_id, version_no) references ezcontentobject_attribute
+                on update cascade on delete cascade
+    );
+
+    CREATE INDEX IF NOT EXISTS ibexa_measurement_range_value_unit_id ON ibexa_measurement_range_value (unit_id);
+
+    CREATE UNIQUE INDEX IF NOT EXISTS ibexa_measurement_range_value_attr_ver
+        on ibexa_measurement_range_value (content_field_id, version_no);
+
+    CREATE UNIQUE INDEX IF NOT EXISTS ibexa_measurement_range_value_attr_ver_type_unit
+        on ibexa_measurement_range_value (content_field_id, version_no, unit_id);
+    ```
+
+### Prepare new database tables
+
+For every database connection you have configured, perform the following steps:
+
+1. Run `php bin/console doctrine:schema:update --dump-sql --em=ibexa_{connection}`
+2. Check the queries and verify that they are safe and will not damage the data.
+3. Run `php bin/console doctrine:schema:update --dump-sql --em=ibexa_{connection} --force`
+
+Run `php bin/console cache:clear` command to make any sure SPI cache issues are cleared.
+
+Run `php bin/console ibexa:migrations:migrate -v --dry-run` to ensure that all migrations are ready to be performed.
+If the dry run is successful, run:
+
+``` bash
+php bin/console ibexa:migrations:migrate
+```
+
+## Update your custom code
+
+### Product catalog
+
+After you update the application and database, account for changes related to the refactored Product Catalog.
+For more information, see [Set up product for purchasing](<link_to_article>).
```

Comment:
Note that the article will only refer to working with 4.1, not to steps necessary for update from 4.0. You need to add update-specific steps from https://github.com/ibexa/product-catalog/pull/342, for example about prices.